docs/design/proposals/logical-volume-manager-storage-api-v2.md

```diff
@@ -0,0 +1,585 @@
+
+Each individual node receives the entirety of the `LVMNodeSet` configuration in it's spec. However, compared to the NodeSet,
+it contains detailed information about `status` elements from `volumeGroups` and `deviceClasses`, where each lvm volume group as well
+as the contents of provisioned deviceClass are checked for consistency and are reported in status. In additions to this,
+the Node contains more detailed conditions that allow setting up detailed status checks
+
+#### `LVMCluster` replaced by `LVMNodeSet`
+
+The current LVMCluster will be broken up into node-specific configurations, wherein each configuration is created through the `LVMNodeSet`,
+which contains a few key spec elements:
+
+1. `spec.nodeSelector` to determine on which nodes to create LVM configurations
+2. `spec.template.volumeGroups` to determine which volumeGroups to create and with which devices based on a `deviceSelector`
+3. `spec.template.deviceClasses` to determine which [TopoLVM device classes](https://github.com/topolvm/topolvm/blob/main/docs/lvmd.md#lvmd) should be prepared and configured for lvm.
+   This also configures a corresponding StorageClass and SnapshotClass for that deviceClass.
+
+```yaml
+---
+apiVersion: lvm.topolvm.io/v2alpha1
+kind: LVMNodeSet
+metadata:
+  name: my-lvmnodeset
+  uid: 31167fec-f8d3-47af-883f-05aaa153da88
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: topology.kubernetes.io/zone
+            operator: In
+            values:
+              - antarctica-east1
+              - antarctica-west1
+      - matchExpressions:
+          - key: custom.topology.domain.io/storage
+            operator: In
+            values:
+              - archive-storage
+  template:
+    volumeGroups:
+      - name: vg1
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: kname
+                  operator: In
+                  values:
+                    - "/dev/mapper/some-encrypted-device"
+                    - "/dev/mapper/some-fallback-device"
+            - matchExpressions:
+                - key: kname
+                  operator: In
+                  values:
+                    - "/dev/mapper/some-default-device"
+      - name: othervg
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: kname
+                  operator: Any
+                - key: size
+                  operator: GreaterThan
+                  values:
+                    - "30Gi"
+      - name: specific-serials
+        deviceSelector:
+          deviceSelectorTerms:
+            - matchExpressions:
+                - key: serial
+                  operator: In
+                  values:
+                    - "2313AW459310"
+    deviceClasses:
+      - name: thinly-provisioned
+        strategy: Thin
+        volumeGroup: vg1
```

Comment:
> Another thought, is it possible to relate multiple vgs to the same thin pool or is it 1-to-1 only?

1-to-1 like you are correctly assuming: A thinpool must exist on exactly one volume group containing n devices. AFAIK it's not possible to create thin-pools from multiple volume groups.

> Can you expand this section to show how other volume groups would also be realted to thin pools here? It's verbose but I think valuable in this situation

What would be your expectation here? Just some more documentation or another volume group being used by a deviceClass? Because of the 1-to-1 relation, im not sure if there is a lot of variance I can add to this example.

Reply:
Maybe just explicitly call out that it's a 1 to 1 relationship somewhere and that should suffice. Just something to make it clear. I can easily see someone who doesn't fully understand thin pools not really understanding that the relationship is 1 to 1

Reply:
Added an explanation that the 1:1 limitation of deviceClasses to volumeGroups is reflecting the same limit in the lvm2 system and is thus not problematic.